benchmarks/_benchmark.py

```diff
@@ -58,13 +58,29 @@ class ConbenchCommunicator(conbench.runner.Conbench):
     publish() method overridden to use the new retrying client.
     """
 
-    def publish(self, benchmark: dict) -> None:
-        # Put login information into environment variables so the new client can access it
+    # Upon first initialization, cache the new retrying client here so we don't have to
+    # login over and over.
+    _conbench_client: Optional[ConbenchClient] = None
+
+    @property
+    def conbench_client(self) -> ConbenchClient:
+        """Set up the new retrying ConbenchClient. And attempt to login.
+
+        (The new client needs login information in environment variables.)
+        """
+        if self._conbench_client:
+            return self._conbench_client
+
         os.environ["CONBENCH_URL"] = self.config.login_url.split("/api/login")[0]
         os.environ["CONBENCH_EMAIL"] = self.config.credentials["email"]
         os.environ["CONBENCH_PASSWORD"] = self.config.credentials["password"]
```

Comment:
just for the record: of course, ugly interface in this context :) (cfg via env, when not needed)

Reply:
agreed! And actually, I just realized we probably don't need this. These environment variables must already be set because we construct the legacy CLI "config file" using them: https://github.com/voltrondata-labs/arrow-benchmarks-ci/blob/6aff3a0027b71ebe32b36a36fc85480f76b0904a/buildkite/benchmark/run.py#L514-L516